src/work.c

```diff
@@ -202,11 +224,17 @@ static int luv_queue_work(lua_State* L) {
     work->args.L = acquire_vm_cb();
   lua_pop(L, 1);
 
-  luv_thread_arg_set(L, &work->args, 2, top, LUVF_THREAD_SIDE_MAIN); //clear in sub threads,luv_work_cb
+  ret = luv_thread_arg_set(L, &work->args, 2, top, LUVF_THREAD_SIDE_MAIN); //clear in sub threads,luv_work_cb
+  if (ret < 0) {
+    luv_thread_arg_clear(L, &work->args, LUVF_THREAD_SIDE_MAIN);
```

Comment:
https://github.com/luvit/luv/blob/b2572ecf988185b9763965c590f6edbb2bc9b6c3/src/work.c#L129-L134

Reply:
This should help

Reply:
@zhaozg  thanks.